src/app/CommandHandler.cpp

```diff
@@ -651,6 +655,9 @@ CHIP_ERROR CommandHandler::FinishCommand(bool aStartDataStruct)
         ReturnErrorOnFailure(commandData.Ref(mRefForResponse.Value()));
     }
 
+    auto * tlvWriter = mInvokeResponseBuilder.GetWriter();
+    ReturnErrorOnFailure(tlvWriter->UnreserveBuffer(mInvokeResponseBuilder.GetInvokeResponses().GetSizeToEndInvokeResponses()));
+    ReturnErrorOnFailure(tlvWriter->UnreserveBuffer(mInvokeResponseBuilder.GetSizeToEndInvokeResponseMessage()));
```

Comment:
This is not making sense to me.  We are neither ending the invoke responses nor the message here.  If we only had one response, this would kind of hang together, but the whole point is we want to have multiple responses, no?

Reply:
You are 100% right here :facepalm: . I copied this to the wrong section. It needed to be in `Finalize`.

Very sorry. I should have caught that